### PR TITLE
[Merged by Bors] - TY-2440 Search events

### PR DESCRIPTION
**Pull request by biernacki**
[Ref](https://xainag.atlassian.net/browse/TY-2440)

### CODE REVIEW

File: discovery_engine/lib/src/api/events/client_events.dart [hunk 1]
```diff
@@ -105,6 +111,29 @@ class ClientEvent with _$ClientEvent {
     UserReaction userReaction,
   ) = UserReactionChanged;
 
+  /// Event created when the user starts a new active search.
+  @Implements<SearchClientEvent>()
+  const factory ClientEvent.searchRequested(
+    String queryTerm,
```
Comment (janpetschexain) [inline discovery_engine/lib/src/api/events/client_events.dart:117]: should we already `assert` in this constructor that the `queryTerm` is not empty? i'd assume that empty querries don't make much sense, if they are even valid querries.

File: discovery_engine/lib/src/discovery_engine_base.dart [hunk 2]
```diff
@@ -268,6 +275,96 @@ class DiscoveryEngine {
     });
   }
 
+  /// Requests a new search for [Document] items related to `queryTerm` for
```
Comment (janpetschexain) [inline discovery_engine/lib/src/discovery_engine_base.dart:278]: similarly below
```suggestion
  /// Requests a new search for [Document]s related to `queryTerm` for
```

File: discovery_engine/lib/src/api/events/engine_events.dart [hunk 3]
```diff
@@ -143,6 +159,54 @@ class EngineEvent with _$EngineEvent {
   const factory EngineEvent.documentsUpdated(List<Document> items) =
       DocumentsUpdated;
 
+  /// Event created as a successful response to SearchRequested event.
+  /// Passes the [ActiveSearch] params and a list of [Document] entities back
+  /// to the client.
+  @Implements<SearchEngineEvent>()
+  const factory EngineEvent.searchRequestSucceeded(
+    ActiveSearch search,
+    List<Document> items,
```
[the remark this answers is not in the document]
Comment (biernacki) [reply discovery_engine/lib/src/api/events/engine_events.dart:168]: maybe at a later time, right now it would be a braking change

**Review by janpetschexain — APPROVED** [review]
lgtm 👍

File: discovery_engine/lib/src/api/events/client_events.dart (hunk 1 above)
Comment (biernacki) [reply discovery_engine/lib/src/api/events/client_events.dart:120]: good idea, so I've added the asserts :)

**Review by acrrd — APPROVED** [review]

**Pull request closed without merging**